File: CS/CustomFunctionCriteriaOperator.Win/WinApplication.cs
using DevExpress.ExpressApp.Xpo;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using DevExpress.ExpressApp.Win;
using DevExpress.ExpressApp.Updating;
using DevExpress.ExpressApp;

using CustomFunctionCriteriaOperator.Module;
using CustomFunctionCriteriaOperator.Module.BusinessObjects;
using DevExpress.Data.Filtering;
using DevExpress.Persistent.Base;
using DevExpress.ExpressApp.Security;

namespace CustomFunctionCriteriaOperator.Win {
   public partial class CustomFunctionCriteriaOperatorWindowsFormsApplication : WinApplication {
        protected override void CreateDefaultObjectSpaceProvider(CreateCustomObjectSpaceProviderEventArgs args) {
            args.ObjectSpaceProvider = new XPObjectSpaceProvider(args.ConnectionString, args.Connection);
        }
      public CustomFunctionCriteriaOperatorWindowsFormsApplication() {
         InitializeComponent();
         DelayedViewItemsInitialization = true;
      }

      private void CustomFunctionCriteriaOperatorWindowsFormsApplication_DatabaseVersionMismatch(object sender, DevExpress.ExpressApp.DatabaseVersionMismatchEventArgs e) {
            e.Updater.Update();
            e.Handled = true;
      }

      private void CustomFunctionCriteriaOperatorWindowsFormsApplication_LastLogonParametersRead(object sender, LastLogonParametersReadEventArgs e) {
         // Just to read demo user name for logon.
         AuthenticationStandardLogonParameters logonParameters = e.LogonObject as AuthenticationStandardLogonParameters;
         if (logonParameters != null) {
            if (String.IsNullOrEmpty(logonParameters.UserName)) {
               logonParameters.UserName = "Sam";
            }
         }
      }
   }
}

File: VB/CustomFunctionCriteriaOperator.Win/WinApplication.vb
Imports DevExpress.ExpressApp.Xpo
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports DevExpress.ExpressApp.Win
Imports DevExpress.ExpressApp.Updating
Imports DevExpress.ExpressApp

Imports CustomFunctionCriteriaOperator.Module
Imports CustomFunctionCriteriaOperator.Module.BusinessObjects
Imports DevExpress.Data.Filtering
Imports DevExpress.Persistent.Base
Imports DevExpress.ExpressApp.Security

Namespace CustomFunctionCriteriaOperator.Win
   Partial Public Class CustomFunctionCriteriaOperatorWindowsFormsApplication
       Inherits WinApplication

        Protected Overrides Sub CreateDefaultObjectSpaceProvider(ByVal args As CreateCustomObjectSpaceProviderEventArgs)
            args.ObjectSpaceProvider = New XPObjectSpaceProvider(args.ConnectionString, args.Connection)
        End Sub
      Public Sub New()
         InitializeComponent()
         DelayedViewItemsInitialization = True
      End Sub

      Private Sub CustomFunctionCriteriaOperatorWindowsFormsApplication_DatabaseVersionMismatch(ByVal sender As Object, ByVal e As DevExpress.ExpressApp.DatabaseVersionMismatchEventArgs) Handles MyBase.DatabaseVersionMismatch
            e.Updater.Update()
            e.Handled = True
      End Sub

      Private Sub CustomFunctionCriteriaOperatorWindowsFormsApplication_LastLogonParametersRead(ByVal sender As Object, ByVal e As LastLogonParametersReadEventArgs) Handles MyBase.LastLogonParametersRead
         ' Just to read demo user name for logon.
         Dim logonParameters As AuthenticationStandardLogonParameters = TryCast(e.LogonObject, AuthenticationStandardLogonParameters)
         If logonParameters IsNot Nothing Then
            If String.IsNullOrEmpty(logonParameters.UserName) Then
               logonParameters.UserName = "Sam"
            End If
         End If
      End Sub
   End Class
End Namespace
